internal/expr/parser.go

```diff
@@ -281,11 +283,21 @@ loop:
 		}
 
 		switch p.input[p.pos] {
-		// If the preceding byte is one of these then we might be at the start
-		// of an expression.
-		case ' ', '\t', '\n', '\r', '=', ',', '(', '[', '>', '<', '+', '-', '*', '/', '|', '%':
-			p.advanceByte()
+		// These bytes may be the start of an expression.
+		case '(', '*', '$', '&':
 			break loop
+		// An expression can also start with an initial name byte, e.g. a
+		// expression starting with a column name or a SQL function.
+		// Rather than testing for every initial name byte (we would stop at
+		// every letter of every word), we look for bytes that may precede the
+		// start of an expression and then check if the next byte is an initial
+		// name byte.
+		case ' ', '\t', '\n', '\r', '=', ',', '[', '>', '<', '+', '-', '/', '|', '%':
+			p.advanceByte()
+			if p.pos >= len(p.input) || isInitialNameByte(p.input[p.pos]) {
```

Comment:
I think that for clarity and for maintenance `p.pos >= len(p.input)` should be a separate clause with an early return. I am afraid that somebody adds code after the loop and it breaks only in the specific instance where the SQL ends with one of these characters. I mean:
```go
if p.pos >= len(p.input) {
     return nil
}
if isInitial... {
    break loop
}